internal/provider/exec_test_data_source.go

```diff
@@ -183,21 +183,24 @@ func (d *ExecTestDataSource) Read(ctx context.Context, req datasource.ReadReques
 	cmd := exec.CommandContext(ctx, "sh", "-c", data.Script.ValueString())
 	cmd.Env = env
 	cmd.Dir = data.WorkingDir.ValueString()
-	out, err := cmd.CombinedOutput()
-	if len(out) > 1024 {
-		out = out[len(out)-1024:] // trim output to the last 1KB
+	fullout, err := cmd.CombinedOutput()
+	if len(fullout) > 1024 {
+		// trim output to the last 1KB
+		trimmedout := fullout[len(fullout)-1024:]
+		data.Output = types.StringValue(string(trimmedout))
+	} else {
+		data.Output = types.StringValue(string(fullout))
 	}
 
 	data.TestedRef = data.Digest
 	data.Id = data.Digest
 	data.ExitCode = types.Int64Value(int64(cmd.ProcessState.ExitCode()))
-	data.Output = types.StringValue(string(out))
 
 	if errors.Is(ctx.Err(), context.DeadlineExceeded) {
 		resp.Diagnostics.AddError("Test timed out", fmt.Sprintf("Test for ref %s timed out after %d seconds", data.Digest.ValueString(), timeout))
```

Comment:
Jason, this is only peripherally related, but I see that we don't include any of the output on timeouts, and I'd love to know what the test was up to for so long: 
![image](https://github.com/chainguard-dev/terraform-provider-oci/assets/2442466/f4f75b78-ae44-4cea-87d5-ba43071b9f70)